### PR TITLE
New package Jali - unstructured mesh infrastructure for multiphysics applications

### PR DESCRIPTION
**Pull request by raovgarimella**
Jali is a parallel unstructured mesh infrastructure library designed for use by multi-physics simulations.

### CODE REVIEW

File: var/spack/repos/builtin/packages/jali/package.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    # dependencies
+    depends_on('cmake@3.13:', type='build')
+
+    depends_on('mpi')
+
+    depends_on('boost')
+
+    depends_on('mstk@3.3.0: +exodusii+parallel~use_markers partitioner=all', when='+mstk')
+
+    depends_on('zoltan -fortran')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jali/package.py]: It won't build against `zoltan+fortran`, or you don't want to by default?

File: var/spack/repos/builtin/packages/jali/package.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    variant('mstk', default=True, description='Enable MSTK')
+
+    # dependencies
+    depends_on('cmake@3.13:', type='build')
+
+    depends_on('mpi')
+
+    depends_on('boost')
+
+    depends_on('mstk@3.3.0: +exodusii+parallel~use_markers partitioner=all', when='+mstk')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jali/package.py]: Are all of these enabled/disabled variants required or preferences?

File: var/spack/repos/builtin/packages/jali/package.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Jali(CMakePackage):
+    """Jali is a parallel unstructured mesh infrastructure library designed for use by multi-physics simulations. It supports 2D and 3D arbitrary polyhedral meshes distributed over hundreds to thousands of nodes. Jali can read and write Exodus II meshes along with fields and sets on the mesh and support for other formats is partially implemented or is in the plans. Jali is built upon MSTK (https://github.com/MeshToolkit/MSTK), an open source general purpose unstructured mesh infrastructure library from Los Alamos National Laboratory. While it has been made to work with other mesh frameworks such as MOAB and STKmesh in the past, support for maintaining the interface to these frameworks has been suspended for now. Jali supports distributed as well as on-node parallelism. Support of on-node parallelism is through direct use of the mesh calls in multi-threaded constructs or through use of "tiles" which are submeshes or sub-partitions of a partition destined for a compute node."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jali/package.py]: Can you wrap this line to 79 characters?

Comment (raovgarimella) [reply var/spack/repos/builtin/packages/jali/package.py]: Done

File: var/spack/repos/builtin/packages/jali/package.py (hunk 2 above)
Comment (raovgarimella) [reply var/spack/repos/builtin/packages/jali/package.py]: For now they are required. We may be able to make some of them variants later.

File: var/spack/repos/builtin/packages/jali/package.py (hunk 1 above)
Comment (raovgarimella) [reply var/spack/repos/builtin/packages/jali/package.py]: It won't build. I get undefined gfortran symbols if I don't disable fortran

File: var/spack/repos/builtin/packages/jali/package.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Jali(CMakePackage):
+    """Jali is a parallel unstructured mesh infrastructure library
+designed for use by multi-physics simulations. It supports 2D and 3D
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/jali/package.py]: Should probably indent this. Also, is this entire description necessary? Most packages have one or two sentences.

Comment (raovgarimella) [reply var/spack/repos/builtin/packages/jali/package.py]: Done.

**Comment by raovgarimella** [comment]
Thanks for the prompt approvals.

**Pull request merged**